geyser-plugin-manager/src/geyser_plugin_service.rs

```diff
@@ -208,4 +150,62 @@ impl GeyserPluginService {
         self.plugin_manager.write().unwrap().unload();
         Ok(())
     }
+
+    fn start_manager_rpc_handler(
+        plugin_manager: Arc<RwLock<GeyserPluginManager>>,
+        request_receiver: Receiver<GeyserPluginManagerRequest>,
+    ) -> Arc<JoinHandle<()>> {
+        Arc::new(thread::spawn(move || loop {
+            if let Ok(request) = request_receiver.recv() {
```

Comment:
You should use recv_timeout with good time our value like a few seconds to check for the exit